tls/s2n_config.c

```diff
@@ -528,11 +528,20 @@ int s2n_config_set_verification_ca_location(struct s2n_config *config, const cha
     return err_code;
 }
 
-static int s2n_config_add_cert_chain_and_key_impl(struct s2n_config *config, struct s2n_cert_chain_and_key *cert_key_pair)
+int s2n_config_add_cert_chain_and_key_impl(struct s2n_config *config, struct s2n_cert_chain_and_key *cert_key_pair)
 {
+    POSIX_ENSURE_REF(config);
+    POSIX_ENSURE_REF(config->security_policy);
     POSIX_ENSURE_REF(config->domain_name_to_cert_map);
     POSIX_ENSURE_REF(cert_key_pair);
 
+    /* Validate that the certificate type is allowed by the security policy. */
+    const struct s2n_security_policy *const security_policy = config->security_policy;
+    if (security_policy->certificate_preferences_apply_locally) {
```

Comment:
nit: Similar comment to https://github.com/aws/s2n-tls/pull/4407/files#r1482447858, it feels strange that `s2n_config_add_cert_chain_and_key_impl` has to know to check `certificate_preferences_apply_locally` before validating the cert chain. It feels like this is a concern of the validate function.